core/src/main/java/org/apache/iceberg/mapping/NameMapping.java

```diff
@@ -27,7 +28,7 @@
 /**
  * Represents a mapping from external schema names to Iceberg type IDs.
  */
-public class NameMapping {
+public class NameMapping implements Serializable {
```

Comment:
While we are using guava classes, I think it should be ok as Spark uses Java serialization for closures.

Reply:
Probably better to switch to using arrays. Some people still use Kryo for task serialization and we don't want that to fail.

Reply:
Thanks! I don't think `Schema` needs to be updated, since it only uses Guava classes for `transient` fields that get re-initialized after serialization.

Reply:
Yeah, you are right, I overlooked.